pyserini/search/__main__.py

```diff
@@ -105,21 +128,39 @@
         output_path = '.'.join(tokens)
 
 print(f'Running {args.topics} topics, saving to {output_path}...')
+tag = output_path[:-4] if args.output is None else 'Anserini'
 
 with open(output_path, 'w') as target_file:
     for index, topic in enumerate(tqdm(sorted(topics.keys()))):
         search = topics[topic].get('title')
         hits = searcher.search(search, args.hits)
-        doc_ids = [hit.docid.strip() for hit in hits]
-        scores = [hit.score for hit in hits]
 
-        if use_prcl and len(hits) > (args.r + args.n):
-            scores, doc_ids = ranker.rerank(doc_ids, scores)
+        if not args.max_passage:
+            docids = [hit.docid.strip() for hit in hits]
+            scores = [hit.score for hit in hits]
+
+            if use_prcl and len(hits) > (args.r + args.n):
+                scores, docids = ranker.rerank(docids, scores)
 
-        if args.msmarco:
-            for i, doc_id in enumerate(doc_ids):
-                target_file.write('{}\t{}\t{}\n'.format(topic, doc_id, i + 1))
+            if args.msmarco:
+                for i, docid in enumerate(docids):
+                    target_file.write(f'{topic}\t{docid}\t{i + 1}\n')
+            else:
+                for i, (docid, score) in enumerate(zip(docids, scores)):
+                    target_file.write(f'{topic} Q0 {docid} {i + 1} {score:.6f} {tag}\n')
         else:
-            tag = output_path[:-4] if args.output is None else 'Anserini'
-            for i, (doc_id, score) in enumerate(zip(doc_ids, scores)):
-                target_file.write(f'{topic} Q0 {doc_id} {i + 1} {score:.6f} {tag}\n')
+            unique_docs = set()
+            rank = 1
+            for hit in hits:
```

Comment:
Yup, `args.max_passage_delimiter` is used here to split `docid#1`. We need it to be a parameter because in TREC-COVID we use `docid.1`.

Reply:
ah, there is a typo then: `max-passage-delimiter` vs `max_passage_delimiter`

Or does argparser automatically convert dashes to underscores?

Reply:
yea, `argparse` converts to underscores automatically because Python variables can't have hyphens.

Reply:
ah ok, thanks!